mozci/task.py

```diff
@@ -200,13 +202,13 @@ def update_group(result):
             ]
 
     def _load_errorsummary(self):
```

Comment:
Could we load/store from the cache outside of _load_errorsummary, just once per push? This way we load/store for every task in a push

Reply:
I have made another comment above. I don't think I've quite managed to accomplish it. Let me know if the idea of parallelization is something you would like me to pursue.

Reply:
You could load the groups directly in `Push.tasks`, and avoid any access to the cache in this function.
We already have some code to parallelize loading groups for all tasks of a push at https://github.com/mozilla/mozci/blob/ef786901e7a74022db0370b5b2a524900d0ef4df/mozci/push.py#L405-L413 (which is likely going to move to a separate function with https://github.com/mozilla/mozci/pull/163/files).

Reply:
By executing that, you'd trigger loading groups for all tasks, once that's done you can perform the caching.